aiohttp/web_protocol.py

```diff
@@ -494,7 +497,9 @@ async def _handle_request(
         else:
             resp, reset = await self.finish_response(request, resp, start_time)
         finally:
-            self._handler_waiter.set_result(None)
+            self._request_in_progress = False
+            if self._handler_waiter is not None and not self._handler_waiter.done():
```

Comment:
We never needed the .done() check before? I don't think that should be possible..

Reply:
Specifically, this method is called below in a `while not self._force_close` loop, and shutdown() sets that to True before making the future. So, there should be no way that it can call this twice while the future exists.

Reply:
removed the done check